routes/web.php

```diff
@@ -16,3 +16,5 @@
 Route::get('/', function () {
     return view('welcome');
 });
+
+Route::resource('contacts', 'ContactController');
```

Comment:
**[Good thing]** the route resource create a pattern in controller code, follow the REST specification and avoid a lot of unnecessary routes here

Reply:
**[warning]** my only suggestion is 

```suggestion
Route::resource('contacts', 'ContactController')->except(['show']);
```
because you don't use the show method